src/org/usfirst/frc/team2175/driverstation/DriverStation.java

```diff
@@ -29,24 +29,50 @@ public DriverStation() {
         leftJoystick = new Joystick(joystickProperties.getJoystickLeftPort());
         rightJoystick = new Joystick(joystickProperties.getJoystickRightPort());
         gamepad = new Joystick(joystickProperties.getGamepadPort());
-        shootButton =
-                new JoystickButton(gamepad, joystickProperties.getRunshooter());
-        shiftButton = new JoystickButton(leftJoystick,
+        shootButton = new JoystickButton(
+                location(joystickProperties.getRunShooterOutLocation()),
+                joystickProperties.getRunShooterOutNumber());
+        feedButton = new JoystickButton(
+                location(joystickProperties.getRunFeederOutLocation()),
+                joystickProperties.getRunFeederOutNumber());
+        shiftButton = new JoystickButton(
+                location(joystickProperties.getShiftButtonLocation()),
                 joystickProperties.getShiftButtonNumber());
-        feederInButton = new JoystickButton(leftJoystick,
-                joystickProperties.getRunFeederIn());
-        shootInButton = new JoystickButton(rightJoystick,
-                joystickProperties.getRunShooterIn());
-        gearIntakeInButton = new JoystickButton(gamepad,
+        feederInButton = new JoystickButton(
+                location(joystickProperties.getRunFeederInLocation()),
+                joystickProperties.getRunFeederInNumber());
+        shootInButton = new JoystickButton(
+                location(joystickProperties.getRunShooterInLocation()),
+                joystickProperties.getRunShooterInNumber());
+        gearIntakeInButton = new JoystickButton(
+                location(joystickProperties.getGearIntakeInLocation()),
                 joystickProperties.getGearIntakeInNumber());
-        gearIntakeOutButton = new JoystickButton(gamepad,
+        gearIntakeOutButton = new JoystickButton(
+                location(joystickProperties.getGearIntakeOutLocation()),
                 joystickProperties.getGearIntakeOutNumber());
 
         deadbandSize = joystickProperties.getDeadbandValue();
         deadbandCalculator = new DeadbandCalculator();
         ServiceLocator.register(this);
     }
 
+    private Joystick location(String str) {
```

Comment:
`location` is not a very good name for this. It really returns a `Joystick`, so maybe something like `joystickForName(String name)` would be better?